src/controllers/board.js

```diff
@@ -120,22 +76,66 @@ export default class BoardController {
 
     const taskListElement = this._tasksComponent.getElement();
 
-    let showingTasksCount = SHOWING_TASKS_COUNT_ON_START;
+    const newTasks = renderTasks(taskListElement, tasks.slice(0, this._showingTasksCount), this._onDataChange, this._onViewChange);
+    this._showedTaskControllers = this._showedTaskControllers.concat(newTasks);
 
-    renderTasks(taskListElement, tasks.slice(0, showingTasksCount));
+    this._renderLoadMoreButton(this._tasks);
+  }
 
-    renderLoadMoreButton();
+  _renderLoadMoreButton(tasks) {
+    if (tasks.length <= this._showingTasksCount) {
+      return;
+    }
 
-    this._sortComponent.setSortTypeChangeHandler((sortType) => {
-      showingTasksCount = SHOWING_TASKS_COUNT_ON_START;
+    const container = this._container.getElement();
+    render(container, this._loadMoreButtonComponent);
 
-      sortedTasks = getSortedTasks(tasks, sortType);
+    this._loadMoreButtonComponent.setLoadMoreButtonClickHandler(() => {
+      const prevTasksCount = this._showingTasksCount;
+      const tasksListElement = this._tasksComponent.getElement();
+      this._showingTasksCount = this._showingTasksCount + SHOWING_TASKS_COUNT_BY_BUTTON;
 
-      taskListElement.innerHTML = ``;
+      const newTasks = renderTasks(tasksListElement, tasks.slice(prevTasksCount, this._showingTasksCount), this._onDataChange, this._onViewChange);
 
-      renderTasks(taskListElement, sortedTasks.slice(0, showingTasksCount));
 
-      renderLoadMoreButton();
+      this._showedTaskControllers = this._showedTaskControllers.concat(newTasks);
+
+      if (this._showingTasksCount >= tasks.length) {
+        remove(this._loadMoreButtonComponent);
+      }
     });
   }
+
+  _onDataChange(taskController, oldData, newData) {
+    const index = this._tasks.findIndex((it) => it === oldData);
+
+    if (index === -1) {
+      return;
+    }
+
+    this._tasks = [].concat(this._tasks.slice(0, index), newData, this._tasks.slice(index + 1));
```

Comment:
```suggestion
    this._tasks = [...this._tasks.slice(0, index), newData, ...this._tasks.slice(index + 1)];
```
Более современный способ. Просто для того чтоб знал)

Reply:
Спасибо.